components/application-gateway/internal/authorization/csrf/csrfclient_test.go

```diff
@@ -0,0 +1,182 @@
+package csrf
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/kyma-project/kyma/components/application-gateway/internal/apperrors"
+	"github.com/kyma-project/kyma/components/application-gateway/internal/authorization"
+	"github.com/kyma-project/kyma/components/application-gateway/internal/authorization/testconsts"
+	"github.com/kyma-project/kyma/components/application-gateway/internal/httpconsts"
+	"github.com/kyma-project/kyma/components/application-gateway/internal/metadata/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	timeoutDuration = 5
+	testURL         = "someURL"
+
+	cachedTestToken            = "someToken"
+	cachedTestCookieName       = "someCookie"
+	endpointTestToken          = "someEndpointToken"
+	endpointResponseCookieName = "someOtherCookie"
+
+	testUsername          = "someUser"
+	testPassword          = "somePassword"
+	expectedAuthHeaderVal = "Basic c29tZVVzZXI6c29tZVBhc3N3b3Jk"
+)
+
+var (
+	certificate = []byte(testconsts.Certificate)
+	privateKey  = []byte(testconsts.PrivateKey)
+)
+
+func TestClient_GetTokenEndpointResponse(t *testing.T) {
+
+	t.Run("Should fetch the token from cache if it is present", func(t *testing.T) {
+
+		// given
+		r := &Response{
+			CSRFToken: cachedTestToken,
+			Cookies:   []*http.Cookie{{Name: cachedTestCookieName}},
+		}
+
+		fakeCache := NewTokenCache()
+		fakeCache.Add(testURL, r)
+
+		c := client{timeoutDuration, fakeCache, nil}
+
+		// when
+		response, appError := c.GetTokenEndpointResponse(testURL, nil)
+
+		// then
+		require.Nil(t, appError)
+		require.NotNil(t, response)
+
+		assert.Equal(t, cachedTestToken, response.CSRFToken)
+		assert.Equal(t, cachedTestCookieName, response.Cookies[0].Name)
+	})
+
+	t.Run("Should fetch the token from endpoint and add it to cache if it is not there", func(t *testing.T) {
+
+		// given
+		fakeCache := NewTokenCache()
+
+		c := client{timeoutDuration, fakeCache, testRequestToken}
```

Comment:
I would suggest different approach to testing this component. In the current approach requestToken function is not tested (it contains the most important part of logic). I think that better approach would be to use httputil.NewServer to start test server so that requestToken will be executed.

Reply:
That sounds reasonable.